earthfile2llb/converter.go

```diff
@@ -291,12 +299,37 @@ func (c *Converter) CopyClassical(ctx context.Context, srcs []string, dest strin
 		c.buildContext, srcs, c.mts.Final.MainState, dest, true, isDir, keepTs, c.copyOwner(keepOwn, chown), false,
 		llb.WithCustomNamef(
 			"%sCOPY %s%s %s",
-			c.vertexPrefix(),
+			c.vertexPrefix(false),
 			strIf(isDir, "--dir "),
 			strings.Join(srcs, " "),
 			dest))
 }
 
+func (c *Converter) RunLocal(ctx context.Context, args []string, pushFlag bool) {
+	runStr := fmt.Sprintf("RUN %s%s", strIf(pushFlag, "--push "), strings.Join(args, " "))
+	finalArgs := withShellAndEnvVars(args, []string{}, true, false)
+	opts := []llb.RunOption{
+		llb.Args(finalArgs),
+		llb.AddMount(localhost.RunOnLocalHostMagicStr, llb.Scratch()), // hack to tell buildkit to run this locally
+		llb.IgnoreCache,
+		llb.WithCustomNamef("%s%s", c.vertexPrefix(true), runStr),
+	}
+
+	if pushFlag {
```

Comment:
Maybe we just don't support push under `FROM LOCAL`? Wonder what the implications for allowing this...

Reply:
Adding support will allow someone to have a bit more control over pushes or not, but they could also just have two separate local targets to support having one perform a push and the other not.

I don't feel strongly attached to it, we can remove it if you think it's a bit too confusing.

Reply:
Perhaps @dchw can look at this and see if it conflicts with his run-after-push work.

Reply:
Nope this is fine. Almost exactly the same as in `RUN`. This is adaptable fairly easily. Were good here.